elastic/kibana
# [Response Ops] Capture durations for different phases in alerting task runner

Resolves https://github.com/elastic/kibana/issues/130134

## Summary

Adds `TaskRunnerTimer` to capture timings of different phases during a rule execution. Open to feedback on the fields names!

* `claim_to_start_duration_ms` - How long between the task `startedAt` and the start of alerting task runner `run()` function. This represents how long it takes the rule to start running after the task is claimed.
* `prepare_rule_duration_ms` - How long it takes the task runner to load and validate the rule saved object
* `rule_type_run_duration_ms` - How long it takes the rule type executor to run.
* `process_alerts_duration_ms` - How long it takes to determine new/active/recovered alerts after rule type execution is done
* `trigger_actions_duration_ms` - How long it takes to schedule all actions
* `process_rule_duration_ms` - How long it takes to finish everything up (reload the rule for the schedule, set execution status, etc)
* `total_run_duration_ms` - How long between the task `startedAt` and the end of the alerting task runner `run()`.

These fields are written out in the event log `execute` document under `kibana.alert.rule.execution.metrics`

## To Verify

1. Create some rules and let them run
2. Go to Dev tools and run this query to get the `execute` events:
```
GET .kibana-event-log-8.5.0/_search
{
  "size": 100,
  "query": {
    "bool": {
      "filter": [
        {
          "terms": {
            "event.action": [
              "execute"
            ]
          }
        }
      ]
    }
  }
}
```
3. Verify that the above listed fields are persisted in `kibana.alert.rule.execution.metrics`
